cargo-lock/tests/lockfile.rs

```diff
@@ -198,155 +198,43 @@ mod tree {
-checksum = "86b6a75fd3048808ef06af5cd79712be8111960adaf89d90250974b38fc3928a"
-dependencies = [
- "bytes 1.2.1 (registry+https://github.com/rust-lang/crates.io-index)",
- "external 1.0.0",
-]
-
-[[package]]
-name = "example"
-version = "0.1.0"
-dependencies = [
- "bytes 0.6.0 (registry+https://github.com/rust-lang/alternate-index)",
- "bytestring",
- "external 2.0.0",
-]
-
-[[package]]
-name = "external"
-version = "1.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "30d69d242d4c4bc978b19d6c5f254cfb61ae3679c4656f528c9992fe337e45a6"
-
-[[package]]
-name = "external"
-version = "2.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "3b5e0d8097cb9529731750ba339ea6813275b868779461ba1d39b841641386d9"
-"#;
-
-    assert_eq!(expected, lockfile);
+fn source_disambiguation_single_version_different_registries() {
+    source_disambiguation("single_version_different_registries");
 }
 
-/// Test that a lockfile with consistent registries are able to skip being encoded
 #[test]
-fn encoding_unified_registry() {
-    let lockfile = r#"# This file is automatically @generated by Cargo.
-# It is not intended for manual editing.
-version = 3
-
-[[package]]
-name = "bytes"
-version = "0.6.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "e0dcbc35f504eb6fc275a6d20e4ebcda18cf50d40ba6fabff8c711fa16cb3b16"
+fn source_disambiguation_single_version_different_source_types() {
+    source_disambiguation("single_version_different_source_types");
+}
 
-[[package]]
-name = "bytes"
-version = "1.2.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "ec8a7b6a70fde80372154c65702f00a0f56f3e1c36abbc6c440484be248856db"
+#[test]
+fn source_disambiguation_two_versions_different_registries() {
+    source_disambiguation("two_versions_different_registries");
+}
 
-[[package]]
-name = "bytestring"
-version = "1.1.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "86b6a75fd3048808ef06af5cd79712be8111960adaf89d90250974b38fc3928a"
-dependencies = [
- "bytes 1.2.1",
-]
+#[test]
+fn source_disambiguation_two_versions_different_source_types() {
+    source_disambiguation("two_versions_different_source_types");
+}
 
-[[package]]
-name = "example"
-version = "0.1.0"
-dependencies = [
- "bytes 0.6.0",
- "bytestring",
-]
-"#;
+#[test]
+fn source_disambiguation_two_versions_same_registry() {
+    source_disambiguation("two_versions_same_registry");
+}
 
-    assert_eq!(
-        lockfile,
-        cargo_lock::Lockfile::from_str(lockfile)
-            .unwrap()
-            .to_string(),
-    );
+/// Test logic related to the disambiguation of dependency specifications when
+/// the lock file contains multiple versions of the same crate and/or multiple
+/// sources for the same crate.
+fn source_disambiguation(test_file_suffix: &str) {
+    let original = std::fs::read_to_string(&format!(
```

Comment:
Nit: suggest importing `fs` in this module.

Reply:
Done (in the first commit).

Reply:
Sidebar: I guess all the examples could use `include_str!`